pkgs/_macro_tool/lib/cfe_macro_runner.dart

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:_cfe_macros/macro_implementation.dart';
+import 'package:front_end/src/macros/macro_injected_impl.dart' as injected_cfe;
+import 'package:frontend_server/compute_kernel.dart';
+import 'package:macro_service/macro_service.dart';
+import 'package:path/path.dart' as p;
+
+import 'macro_runner.dart';
+import 'source_file.dart';
+
+class CfeMacroRunner implements MacroRunner {
+  final String workspacePath;
+  final String packageConfigPath;
+
+  @override
+  final List<SourceFile> sourceFiles;
+
+  CfeMacroImplementation? cfeMacroImplementation;
+
+  CfeMacroRunner({required this.workspacePath, required this.packageConfigPath})
+      : sourceFiles = Directory(workspacePath)
```

Comment:
This looks like it could move into the `MacroRunner` class and then you would extend it from the CFE/analyzer.

The workspace path and packageConfigPath could also become concrete fields in that class.